projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-row.component.html

```diff
@@ -1,4 +1,4 @@
-<div (click)="toggle()" class="igx-grid__hierarchical-expander" [tabIndex]="tabindex" *ngIf="hasChildren" #expander>
+<div (click)="expanderClick($event)" class="igx-grid__hierarchical-expander" [tabIndex]="tabindex" *ngIf="hasChildren" #expander>
```

Comment:
Instead of creating a new method, just add the logic to the template:

`(click)="toggle(); $event.stopPropagation()"`

Reply:
@kdinev I'd rather not. Having multiple function calls inside the template seems like a bad practice. Besides, the reason this got regressed is a bad merge due to conflicts in the template file, where the merge tool doesn't easily pick up small changes in the html.